### PR TITLE
feat: [CDS-42091]: Service V2 support for Jenkins API

### PR DESCRIPTION
**Pull request by shivamnegi94**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37936)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 8 Issues 
* Code issue 5 Code issues 
* Anti-pattern 3 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37936)

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/jenkins/JenkinsArtifactResource.java [hunk 1]
```diff
@@ -119,4 +125,134 @@ public ResponseDTO<List<JobParameter>> getJobParameters(@QueryParam("connectorRe
...
+      @QueryParam("parentJobName") String parentJobName, @BeanParam GitEntityFindInfoDTO gitEntityBasicInfo,
+      @NotNull @QueryParam("fqnPath") String fqnPath,
+      @QueryParam(NGCommonEntityConstants.SERVICE_KEY) String serviceRef, @NotNull String runtimeInputYaml) {
+    if (isNotEmpty(serviceRef)) {
+      final ArtifactConfig artifactSpecFromService = artifactResourceUtils.locateArtifactInService(
+          accountId, orgIdentifier, projectIdentifier, serviceRef, fqnPath);
+      JenkinsArtifactConfig jenkinsArtifactConfig = (JenkinsArtifactConfig) artifactSpecFromService;
+      if (isEmpty(jenkinsConnectorIdentifier)) {
+        jenkinsConnectorIdentifier = jenkinsArtifactConfig.getConnectorRef().getValue();
+      }
```
Comment (hingerabhinav) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/jenkins/JenkinsArtifactResource.java:146]: parentJobName would be available always? Should we get from config if its empty in request?

**Review by hingerabhinav — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37936)* status: 1 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/jenkins/JenkinsArtifactResource.java` line 137 at r2](https://reviewable.io/reviews/harness/harness-core/37936#-ND2T_yR0aWlE-4C2ktr:-ND2T_yS8GlLNy4MYRGT:b-g9eghl) ([raw file](https://github.com/harness/harness-core/blob/b4ed33bd419cd98810c6cb39e09711ef65a22892/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/jenkins/JenkinsArtifactResource.java#L137)):*
> ```Java
>       @NotNull @QueryParam(NGCommonEntityConstants.ORG_KEY) String orgIdentifier,
>       @NotNull @QueryParam(NGCommonEntityConstants.PROJECT_KEY) String projectIdentifier,
>       @QueryParam("parentJobName") String parentJobName, @BeanParam GitEntityFindInfoDTO gitEntityBasicInfo,
> ```

Let's create constants for these params. We can reuse them.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/jenkins/JenkinsArtifactResource.java` line 146 at r2](https://reviewable.io/reviews/harness/harness-core/37936#-NCuWZcJ_WD4wuYAl7NC-r2-146:-ND2TemR1h_gmXa4ilbW:b2kv38v) ([raw file](https://github.com/harness/harness-core/blob/b4ed33bd419cd98810c6cb39e09711ef65a22892/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/jenkins/JenkinsArtifactResource.java#L146)):*
<details><summary><i>Previously, shivamnegi94 (Shivam Negi) wrote…</i></summary><blockquote>

ParentJob name is not in yaml, UI have to provide this value
</blockquote></details>

Got it. Can it be given as an expression? Then we will need resolving logic here as well.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/jenkins/JenkinsArtifactResource.java` line 178 at r2](https://reviewable.io/reviews/harness/harness-core/37936#-ND2UNFnD4-wPXVimwUF:-ND2UNFnD4-wPXVimwUG:bygpwf5) ([raw file](https://github.com/harness/harness-core/blob/b4ed33bd419cd98810c6cb39e09711ef65a22892/120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/jenkins/JenkinsArtifactResource.java#L178)):*
> ```Java
>       }
>     }
>     jobName = artifactResourceUtils.getResolvedImagePath(accountId, orgIdentifier, projectIdentifier,
> ```

Please create a tech-debt to refactor this method since it works with other fields other than image path.



_Code quote:_
```Java
getResolvedImagePath
```

**Review by puthrayaharness — DISMISSED** [review]

**Review by hingerabhinav — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37936)* status: :shipit: complete! ✓

**Pull request merged**